api/repo.go

```diff
@@ -90,87 +91,90 @@ func CreateRepo(c *gin.Context) {
 		return
 	}
 
+	// get repo information from the source
+	r, err := source.FromContext(c).GetRepo(u, input)
```

Comment:
This is the main change. After we parse the given input, we get the repo information from the source. We will use the returned repo struct as a building block for what will eventually be activated.